### PR TITLE
Fix: defer in loop will cause resource leak

### PR DESCRIPTION
**Pull request by watermelo**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: config/service_config.go [hunk 1]
```diff
@@ -119,8 +119,8 @@ func getRandomPort(protocolConfigs []*ProtocolConfig) *list.List {
 		if err != nil {
 			panic(perrors.New(fmt.Sprintf("Get tcp port error,err is {%v}", err)))
 		}
-		defer tcp.Close()
 		ports.PushBack(strings.Split(tcp.Addr().String(), ":")[1])
+		tcp.Close()
```
Comment (zouyx) [inline config/service_config.go:123]: I thought may panic in line 122 ,so use defer in loop .
In this case ,  resoure usage will under control . 
what do you think?

Comment (flycash) [inline config/service_config.go:123]: I am not very sure whether `tcp.Addr().String()` will return something like "127.0.0.1" which will cause `ports.PushBack(strings.Split(tcp.Addr().String(), ":")[1])` panic.

Comment (watermelo) [reply config/service_config.go]: updated.

Comment (watermelo) [reply config/service_config.go]: updated.

**Pull request closed without merging**